asv/benchmark.py

```diff
@@ -1151,17 +1157,21 @@ def main_run(args):
         if cache is not None:
             benchmark.insert_param(cache)
 
-    skip = benchmark.do_setup()
-
     try:
-        if skip:
-            result = float('nan')
-        else:
-            result = benchmark.do_run()
-            if profile_path is not None:
-                benchmark.do_profile(profile_path)
-    finally:
-        benchmark.do_teardown()
+        skip = benchmark.do_setup()
+
+        try:
+            if skip:
+                result = float('nan')
+            else:
+                result = benchmark.do_run()
+                if profile_path is not None:
+                    benchmark.do_profile(profile_path)
+        finally:
+            benchmark.do_teardown()
+    except BaseException as e:
```

Comment:
`finally:` I think

Reply:
Using `finally` would mean resorting to using `sys.exc_info()` to get the exception info and checking if it differs from `(None, None, None)`, which to me feels more convoluted than just using an `except` clause.

Reply:
Yes, but the point is that the finalize should run always, not only for exceptions.